rfcs/012-add-session-ssh-support/012-add-session-ssh-support.md

```diff
@@ -0,0 +1,438 @@
+ssh servers running inside user sessions directly to the Internet brings some
+security risks and it would need to be clear that users cannot easily change
+the ssh daemon configuration within their sessions to make their session very
+exposed.
+
+A couple of further points relating to this approach:
+- ingresses on cloud providers can typically be expensive; having one per
+  session is likely to incur significant cost;
+- on the Switch deployments, the entire cluster is exposed via a single IP
+  address; as such it is not straightforward to devise a solution in which
+  different sessions would map to different IP addresses.
+
+### Adding a proxy jumphost
+
+Another solution and that which is currently deployed is to have a jumphost and
+to use this to access the pods. In this case, a jumphost must be installed
+inside the kubernetes cluster as it requires direct access to the ssh server
+running on the pods. The jumphost has a standard ssh server running with proxying
+enabled.
+
+For the Switch/Openstack deployments, this requires modifications to the
+cluster ingress configuration to forward TCP traffic incident on a specific
+port to the jumphost pod on its ssh port. For cloud providers, a new ingress
+pointing at the ssh proxy is straightforward. ssh proxying approaches require
+authentication against both the proxy and the destination server; this
+necessitates an approach in which some combination of no credentials and a
+keypair must be deployed to the proxy and/or user session.
+
+The most natural solution in this case is to use the user's public key within
+the session and have open access with no ability to obtain a shell within the
+jumphost. Key management then becomes a user responsibility, with users having
+to do this within their Renku projects; as such, keys are project entities,
+rather than user entities.
+
+In this case, the command to ssh into a session requires specifying both the
+jumphost and the destination; hence it is a more complex command and introduces
+some user friction when accessing sessions. VSCode supports such proxying
+scenarios without any problems.
+
+### Using a Man In the Middle (MITM) Proxy solution
+
+The third approach is the one followed by gitpod; in this approach a dedicated
+proxy is provided which affords access to the sessions. As with the proxying
+approach above, this must run on the kubernetes cluster.
+
+Unlike the above approach, this requires writing and maintaining a new software
+component. The key difference is that this proxy does not use the standard ssh
+proxying capabilities provided by an ssh client (in which one ssh session is
+essentially embedded inside another); effectively this approach concatenates
+two distinct ssh sessions with two distinct authentication flows. The proxy
+terminates one ssh session, relaying all traffic to another ssh session.
+
+This solution has the following benefits:
+- keys can be clearly bound to users and separated from projects
```

Comment:
Again, see my point above - the current implementation with the jumphost does not technically require that the keys are inside the project; it was simply the only way to do it with "what we have".